src/main/kotlin/com/ampnet/payoutservice/util/ValueClasses.kt

```diff
@@ -39,8 +47,12 @@ value class ChainId(val value: Long)
 value class BlockNumber(val value: BigInteger)
 
 @JvmInline
-value class ContractAddress(val value: Address) {
-    constructor(value: String) : this(Address(value))
+value class ContractAddress private constructor(val value: Address) {
+    companion object {
+        operator fun invoke(value: Address) = ContractAddress(value.toString())
+    }
+
+    constructor(value: String) : this(Address(value.lowercase()))
```

Comment:
This is excellent because we usually had problems with the same addresses but some were in lowercase and some in uppercase.

Reply:
Thanks 😄  I will also do this in other services when I start with SD-258